src/main/java/com/powsybl/openloadflow/ac/ReactiveLimitsOuterLoop.java

```diff
@@ -196,10 +194,12 @@ private void checkPqBus(LfBus controllerCapableBus, List<PqToPvBus> pqToPvBuses)
         double minQ = controllerCapableBus.getMinQ();
         double maxQ = controllerCapableBus.getMaxQ();
         double q = controllerCapableBus.getGenerationTargetQ();
-        if (Math.abs(q - maxQ) < Q_EPS && controllerCapableBus.getV().eval() > getBusTargetV(controllerCapableBus)) { // bus produce too much reactive power
+        double distanceToMaxQ = Math.abs(q - maxQ);
+        double distanceToMinQ = Math.abs(q - minQ);
+        if (distanceToMaxQ < distanceToMinQ  && controllerCapableBus.getV().eval() > getBusTargetV(controllerCapableBus)) { // bus produce too much reactive power
```

Comment:
@geofjamg can you please validate these new lines. For me, it works because we are sure that the bus checked is a PQ bus that where previously PV. Do you agree?